rclpy/rclpy/executors.py

```diff
@@ -717,3 +733,8 @@ def spin_once(self, timeout_sec: float = None) -> None:
             pass
         else:
             self._executor.submit(handler)
+
+    def spin_once_until_future_complete(self, future: Future, timeout_sec: float = None) -> None:
+        self.spin_once(timeout_sec)
+        if future.done():
```

Comment:
future might not be done yet in child thread, then main thread will call in this function, wait on `rcl_wait`.